### PR TITLE
[Storage] [STG93] Support owner/group/permission/acl on put APIs and `x-ms-upn` header

### PR DESCRIPTION
**Pull request by vincenttran-msft**
TODO:

- Need to flesh out a `x-ms-upn` test when we can do it reasonably
- Need to figure out all the remaining APIs that it is supposed to be on (currently only PoC on `get_file_properties` & `get_directory_properties`)

### CODE REVIEW

File: sdk/storage/azure-storage-file-datalake/tests/test_file.py [hunk 1]
```diff
@@ -1569,15 +1569,17 @@ def test_dir_and_file_properties_owner_group_permissions(self, **kwargs):
...
 
         directory_properties = directory_client.get_directory_properties()
-        file_properties = file_client1.get_file_properties()
+        file_properties = file_client1.get_file_properties(upn=True)
 
         # Assert
         assert directory_properties['owner'] is not None
         assert directory_properties['group'] is not None
         assert directory_properties['permissions'] is not None
+        assert directory_properties['acl'] is not None
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-file-datalake/tests/test_file.py:1578]: Did you mean to add `upn=True` to `get_directory_properties`? Or did we want one with and one without?

Likely when we actually test this we'll need a new test anyway.

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_download.py [hunk 2]
```diff
@@ -26,9 +26,13 @@ def __init__(self, downloader):
         self.name = self._downloader.name
 
         # Parse additional Datalake-only properties
-        encryption_context = self._downloader._response.response.headers.get('x-ms-encryption-context')  # pylint: disable=line-too-long,protected-access
+        encryption_context = self._downloader._response.response.headers.get('x-ms-encryption-context')  # pylint: disable=line-too-long, protected-access
+        acl = self._downloader._response.response.headers.get('x-ms-acl')  # pylint: disable=line-too-long, protected-access
```
[the remark this answers is not in the document]
Comment (vincenttran-msft) [reply sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_download.py:30]: True for acl, not for encryption_context 😛

File: sdk/storage/azure-storage-file-datalake/tests/test_file.py (hunk 1 above)
Comment (vincenttran-msft) [reply sdk/storage/azure-storage-file-datalake/tests/test_file.py:1578]: Test cases are not written yet -- will round back to this
As for `get_directory_properties`, I just added it there. I am not exactly sure if it should apply (I feel like it should) but the spec isn't exactly 100% clear. If anything this is an easy revert after archboard feedback

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_directory_client.py [hunk 3]
```diff
@@ -301,6 +310,11 @@ def get_directory_properties(self, **kwargs):
                 :dedent: 4
                 :caption: Getting the properties for a file/directory.
         """
+        upn = kwargs.pop('upn', None)
+        if upn:
+            headers = kwargs.pop('headers', {})
+            headers['x-ms-upn'] = str(upn)
+            kwargs['headers'] = headers
         return self._get_path_properties(cls=deserialize_dir_properties, **kwargs)  # pylint: disable=protected-access
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_directory_client.py:318]: So now that directory is included, I think you can move the `upn` logic into `_get_path_properties` of `PathClient` (and async). That way its shared for file/directory.

**Pull request merged**